confd/templates/opennms-activemq.xml.tpl

```diff
@@ -0,0 +1,195 @@
+            JMX. By default, ActiveMQ uses the MBean server that is started by
+            the JVM. For more information, see:
+
+            http://activemq.apache.org/jmx.html
+        -->
+        <managementContext>
+            <managementContext createConnector="false"/>
+        </managementContext>
+
+        <!--
+            Configure message persistence for the broker. The default persistence
+            mechanism is the KahaDB store (identified by the kahaDB tag).
+            For more information, see:
+
+            http://activemq.apache.org/persistence.html
+        -->
+        <persistenceAdapter>
+            <kahaDB directory="${activemq.data}/kahadb"/>
+        </persistenceAdapter>
+
+
+          <!--
+            The systemUsage controls the maximum amount of space the broker will
+            use before slowing down producers. For more information, see:
+            http://activemq.apache.org/producer-flow-control.html
+            If using ActiveMQ embedded - the following limits could safely be used:
+        -->
+        <systemUsage>
+            <systemUsage>
+                <memoryUsage>
+                    <memoryUsage limit="{{getv "/opennms/activemq/systemusage/memoryusage" "20 mb"}}"/>
+                </memoryUsage>
+                <storeUsage>
+                    <storeUsage limit="{{getv "/opennms/activemq/systemusage/storageusage" "1 gb"}}"/>
+                </storeUsage>
+                <tempUsage>
+                    <tempUsage limit="{{getv "/opennms/activemq/systemusage/tempusage" "100 mb"}}"/>
+                </tempUsage>
+            </systemUsage>
+        </systemUsage>
+
+        <!--
+            The transport connectors expose ActiveMQ over a given protocol to
+            clients and other brokers. For more information, see:
+
+            http://activemq.apache.org/configuring-transports.html
+        -->
+        <transportConnectors>
+            <transportConnector name="vm" uri="vm://localhost"/>
+            <!-- Uncomment this line to allow external TCP connections -->
+            <!--
+              WARNING: Access to port 61616 should be firewalled to prevent unauthorized injection
+              of data into OpenNMS when this port is open.
+            -->
+            <!-- <transportConnector name="openwire" uri="tcp://0.0.0.0:61616?useJmx=false&amp;maximumConnections=1000&amp;wireformat.maxFrameSize=104857600"/> -->
+
+            <!-- Uncomment this line to allow localhost TCP connections (for testing purposes) -->
+            <!-- <transportConnector name="openwire" uri="tcp://127.0.0.1:61616?useJmx=false&amp;maximumConnections=1000&amp;wireformat.maxFrameSize=104857600"/> -->
+            {{if exists "/opennms/activemq/transportconnectors/openwire/uri"}}
+            <transportConnector name="openwire" uri="{{getv "/opennms/activemq/transportconnectors/openwire/uri"}}/>
+            {{end}}
+        </transportConnectors>
```

Comment:
I don't follow the reason why defining a full URI for openwire .... considering what's documented to use either an internal AMQ broker or an external AMQ broker.

What I found that works better is to just remove the comments around the *existing* `transportConnector` associated with `0.0.0.0` only when `org.opennms.activemq.broker.disable=false` (assuming the local AMQ broker will be used), and just never touch that file when using an external broker is defined, which requires `org.opennms.activemq.broker.disable=true` for an external AMQ, or when using a different technology like Kafka.

On a side note, I found there are performance improvements when disabling AMQ if it is not going to be used (like an OpenNMS without Minions involved).